src/components/events/EventItem.js

```diff
@@ -5,23 +5,46 @@ import {
   capitalizeFirstLetter,
   getDayColors,
 } from "../../utils/common";
+import { CautionIcon, TimerIcon, LocationIcon } from "../icons/Icons";
 
-export const EventItem = ({ eventData }) => {
+export const EventItem = ({ eventData, onClickReport, setReportedDeal, setReportedPlace }) => {
```

Comment:
I think we want to pull the report modal into this component itself and then open it when the button is clicked. We don't want to pass the state setters and onClickreport to here. If we were to not move the modal then we can have the onClickreport function take the place id and deal id